### DISCUSSION TITLE
Debug on debug console

### DISCUSSION THREAD

**Discussion by ECEAthipanbose** [scalameta/metals-vscode#1459] [Q&A] [answered]
When debugging Scala code, i need to receive input from the user via the debug console, when step over the (val name = StdIn.readLine()) code , i can't directly interact with the debug console to provide input and debugger not move remaining line

**Comment by tgodzik** [comment]
That's a limitation of debug console in VS Code. You would need to attach to a running process instead with the additional debug flags.

**Reply by ECEAthipanbose** [reply]
how can we open the terminal without using debug console? we need to recieve user input

**Reply by tgodzik** [reply]
debug console is for interacting with the debugger, not with the user input. You need to run a program in the console and attach to it later on. There is a separate configuration in VS Code that you can use for that and it should also have an example of the properties to use.

**Reply by ECEAthipanbose** [reply]
but when i debug my scala program the debugger not step over the remianing code when it reach the code line(like val name = StdIn.readLine())
eg:-
object Sample {
  def main(args: Array[String]) = {
    var s = new Student() // Creating an object
    println(s.id + " " + s.name);
        // Prompt the user for input
    print("Enter your name: ")
    // Read input from the user  
    val name = StdIn.readLine() 
    // Display the input
    println(s"Hello, $name! Welcome to Scala programming.") 
  }       
}   
example when i debug the above code it work fine and reach the code val name = StdIn.readLine() now i step over the debugger but it not move the code to
 println(s"Hello, $name! Welcome to Scala programming.")

**Reply by tgodzik** [reply]
Most likely it doesn't move, because the program doesn't get any input. Which is why you would need to start the program separately and then attach to it. There is separate template for attaching with Metals if you write `scala...` in `.vscode/launch.json`

**Reply by ECEAthipanbose** [reply]
thak you  tgodzik !

there is any doc available for explain the concept( how start the program separately and then attach to it.
)?

**Reply by tgodzik** [reply, marked as answer]
https://www.baeldung.com/java-application-remote-debugging
